autolens/lens/ray_tracing.py

```diff
@@ -399,6 +401,18 @@ def save(self, file_path, filename="tracer"):
 
 
 class AbstractTracerLensing(AbstractTracer, ABC):
+    def __init__(self, planes, cosmology, profiling_dict: Optional[Dict] = None):
+
+        super().__init__(
+            planes=planes, cosmology=cosmology, profiling_dict=profiling_dict
+        )
+
+        self._calc_image = CalcImage(image_2d_from=self.image_2d_from)

```

Comment:
Consider whether the Calc* classes need to be instantiated in here and whether the methods need to be copied over. What if we instantiate the Calc* objects where they're used instead?

Calc* and Tracer are very tightly coupled.